src/main/java/svc/data/citations/datasources/rejis/RejisCitationDataSource.java

```diff
@@ -0,0 +1,211 @@
+	}
+	
+
+	private UriComponentsBuilder getNameBuilder(int pageNumber, String lastName, LocalDate dob, List<String> municipalityCodes){
+		UriComponentsBuilder builder = UriComponentsBuilder.fromHttpUrl(rejisConfiguration.getRootUrl()+"/byName")
+				.queryParam("LastName", lastName)
+				.queryParam("DobYear", dob.format(dobFormatter))
+				.queryParam("AgcyIdOri",getMunicipalitiesString(municipalityCodes))
+				.queryParam("PageNum", pageNumber)
+				.queryParam("RowsPerPage", 50)
+				.queryParam("ResultFormat","json");
+		
+		return builder;
+	}
+
+	private HttpEntity<?> getHttpEntity(){
+		HttpHeaders headers = new HttpHeaders();
+		headers.add("Authorization", rejisConfiguration.apiKey);
+		return new HttpEntity<>(headers);
+	}
+	
+	private RejisCaseList performRestTemplateCall(URI uri) {
+		RejisCaseList rejisCaseList = null;
+		
+		try{
+			ResponseEntity<RejisCaseList> response = restTemplate.exchange(uri, HttpMethod.GET, getHttpEntity(), RejisCaseList.class);
+			rejisCaseList = response.getBody();
+		}catch(Exception e){
+			LogSystem.LogCitationDataSourceException("Rejis datasource exception: " + e.getMessage());
+		}
+		
+		return rejisCaseList;
+	}
+	
+	private List<Citation> getFullCitations(List<RejisPartialCitation> partialCitations){
+		List<Observable<Citation>> citations = Lists.newArrayList();
+		for(RejisPartialCitation partialCitation: partialCitations){
+			citations.add(Observable.from(getCitation(partialCitation)).onExceptionResumeNext(Observable.just(null)));
+		}
+		return Observable.merge(citations).onExceptionResumeNext(Observable.just(null)).toList().toBlocking().first();
+	}
+	
+	private RejisFullCitation getFullCaseView(RejisPartialCitation partialCitation){
+		UriComponentsBuilder builder = UriComponentsBuilder.fromHttpUrl(rejisConfiguration.getRootUrl()+"/CaseView")
+				.queryParam("AgcyIdOri",partialCitation.AgcyId)
+				.queryParam("CaseNumber", partialCitation.CaseNum);
+		
+		RejisFullCitation rejisFullCitation = null;
+		
+		try{
+			ResponseEntity<RejisFullCitation> response = restTemplate.exchange(builder.build().encode().toUri(), HttpMethod.GET, getHttpEntity(), RejisFullCitation.class);
+			rejisFullCitation = response.getBody();
+		}catch(Exception e){
+			LogSystem.LogCitationDataSourceException("Rejis datasource exception: " + e.getMessage());
+		}
+		
+		return rejisFullCitation;
+	}
+	
+	private List<Citation> getCitation(RejisPartialCitation partialCitation){
+		RejisFullCitation rejisFullCitation = getFullCaseView(partialCitation);
+		if (rejisFullCitation == null){
+			return Lists.newArrayList();
+		}
+		Citation citation = citationTransformer.fromRejisFullCitation(rejisFullCitation, partialCitation);
+		return Arrays.asList(citation);
+	}
+	
+	private String getMunicipalitiesString(List<String> municipalityCodes){
+		StringJoiner joiner  = new StringJoiner(",");
+        for(String muni : municipalityCodes) { joiner.add(muni); }
+        
+        return joiner.toString();
+	}
+}
```

Comment:
This thing is 211 lines long....way too long for new code. Sometimes code will grow to this size if it makes sense but we shouldn't purposefully make a class this big. I think a clear division lies where you can make a new class to actually construct the REJIS API requests and execute the requests that this Data Source calls into. Then the Data Source can package up the response however it needs (i.e. transform, normalize, etc.)